### DISCUSSION TITLE
How to automatically switch to Class B when Class B is lost

### DISCUSSION THREAD

**Discussion by 13868132273** [Lora-net/LoRaMac-node#1186] [Q&A] [answered]
I have some questions about Class B
It will be much appreciate if I can get help from you

Class B when the beacon is lost for more than 2 hours, it will switch to Class A. how can I automatically switch to Class B again ?
This step is not found in the code, only in MLME_BEACON_LOST has an LmHandlerDeviceTimeReq().

best regards !

**Comment by mluis1** [comment, marked as answer]
Once, this situation happens you can follow the same procedure as the one used for the first time.
Calling the `LmHandlerRequestClass` API with `CLASS_B` as parameter will restart the process to switch to Class B again.


In the future it would be nice if you could post this kind of questions on the project [Discussions](https://github.com/Lora-net/LoRaMac-node/discussions) tab. It is a better place to engage discussions and then we can agree if it is an issue or not.
If you don't mind I move this issue to the [Discussions](https://github.com/Lora-net/LoRaMac-node/discussions) tab.

**Comment by 13868132273** [comment]
I'll keep in mind to post it in the discussions tab next time.
Thank you so much @mluis1.
